### DISCUSSION TITLE
Simple way to split failing dependency PRs into smaller packages or even individual ones?

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#14628] [Request Help]
We usually have a `dependencies (major)` group that we only sometimes look into. Often that has collected a few dependencies over time and is failing. Is there a simple way to split up that failing PR into multiple smaller ones so it is easier to figure out which dependency update is causing the issue?

Or an alternative approach: When I know which dependency is causing a group PR to fail, can I somehow easily exclude that update so the PR can pass and be merged, and the failing one is split into its own PR that we can look into individually?

Thanks.

**Comment by wolfgangwalther** [comment]
Yes, that would be a great feature to have. Something like: When a PR with multiple dependencies fails, split the dependencies in half and create two separate PRs. Repeat until possibly a single dependency is left in one PR, and all other PRs succeed.

Or alternatively: Add a checkmark to each dependency in the list of deps to split this off into their own PR just for this update - without having to change the renovate config file in general. This would allow manual inspection of failure, splitting off the failing update and merging the passing deps quickly.

**Comment by rarkins** [comment]
1. Grouping major updates together - unless they are from the same monorepo or project - is nearly always going to cause problems
2. I would prefer to reverse the problem, by first testing them separately and then "rolling them up" into a combined PR so that you get only the passing ones: https://github.com/renovatebot/renovate/issues/4404

**Reply by MartijnLeijssen-TomTom** [reply]
I would love to have this feature over (next to) roll up as this feature is fixing different practical issues.  In the use case I have, we use renovate to propagate a large amount  of internal dependencies to a product, the amount of new versions is of all the components is quite big as the components are all in full development. We are looking to improve the intake with the following criteria: 
- Increase visibility on which component breaks intake
- Merge non breaking components as fast as possible

The build time / resource usage of building the components shouldn't be underestimated in our case. It might be relevant to mention that we have a C++ stack, which means in practice that all components that depend on the component being changed will be rebuilt (from scratch). With roll-up the idea is to build every single change up front (if I understand correctly), which means that combining two changes already gives a entire new composition, meaning that the potentially the whole stack has to be rebuild.

We have enabled auto-merge, so in case roll-up would be used and a single component build would be green, I wouldn't mind to merge it directly. Testing them separately (roll-up) will improve the first criteria I mentioned, but given the significant build/test time I would rather try to build the new composition first and split in case of a failure
